rest_api/limsdb/queries/data_models.py

```diff
@@ -153,7 +153,9 @@ def additional_status(self):
         for process, date, process_type, process_id in self.processes:
             if process_type == 'complete' and process in status_cfg.additional_step_completed:
                 finished_status = status_cfg.additional_step_completed.get(process)
-                additional_status.add(finished_status)
+                # A sample should not appear in invoiced if it has been removed, as this is paradoxical
+                if self.status != 'removed' or finished_status == 'genotyped':
```

Comment:
I think I would change the `finished_status == 'genotyped'` by `finished_status != 'invoiced'`.
It has the advantage to match the comment and be more future proof if a new additional status gets added.